### PR TITLE
Update devDependencies and drop Node.js < 10 support

### PR DESCRIPTION
**Pull request by XhmikosR**
@kristerkari feel free to push to this branch before merging to fix the remaining lint issues and ping me. Note that this PR targets the v4 branch.

Non-whitespace diff: <https://github.com/kristerkari/stylelint-scss/pull/469/files?w=1>

### CODE REVIEW

**Comment by XhmikosR** [comment]
I disabled the jest-setup.js errors. There are only 2 left in `/src/utils/__tests__/parseFunctionArguments.js`

File: package.json [hunk 1]
```diff
@@ -32,61 +32,31 @@
     "stylelint": "^8.0.0 || ^9.0.0 || ^10.0.0 || ^11.0.0 || ^12.0.0 || ^13.0.0"
   },
   "engines": {
-    "node": ">=8"
+    "node": ">=10"
```
Comment (XhmikosR) [inline package.json:35]: BTW if you want to make this stricter let me know. We should also change the version we test on CI to match the minimum one.

File: package.json [hunk 2]
```diff
@@ -32,61 +32,31 @@
...
-      "template-curly-spacing": "error",
       "lodash/callback-binding": "error",
       "lodash/collection-method-value": "error",
       "lodash/collection-return": "error",
       "lodash/no-double-unwrap": "error",
       "lodash/no-extra-args": "error",
       "lodash/no-unbound-this": "error",
       "lodash/unwrap": "error",
-      "lodash/preferred-alias": "error",
-      "node/no-unsupported-features/es-syntax": [
```
Comment (XhmikosR) [inline package.json:81]: This is inherited by the stylelint eslint config. We could keep it and adapt it if you prefer that.

File: package.json (hunk 1 above)
Comment (kristerkari) [reply package.json:35]: I think that this is good enough.

File: package.json (hunk 2 above)
Comment (kristerkari) [reply package.json:81]: Sorry, what do you mean with "this"?

Comment (XhmikosR) [reply package.json:81]: This rule.

**Comment by kristerkari** [comment]
> I disabled the jest-setup.js errors. There are only 2 left in `/src/utils/__tests__/parseFunctionArguments.js`

Yeah I can fix the current warnings/errors.

File: package.json (hunk 2 above)
Comment (kristerkari) [reply package.json:81]: maybe we should keep it if we want to have a separate minimum node version from stylelint. It should not be a big problem if we inherit it, but we could run into some compatibility issues at some point.

**Comment by XhmikosR** [comment]
@kristerkari maybe we should silence the jest/no-standalone-expect errors in jest-setup.js?

Feel free to fix those too. And also, if you want me to cherry pick any of these patches in master let me know. I can rebase this branch before the final merge.

File: package.json (hunk 2 above)
Comment (XhmikosR) [reply package.json:81]: I'll have a look later an restore this.

**Comment by XhmikosR** [comment]
Oh, and also, it would be nice if we could find a way to fix these warnings too :)

>You did not set any plugins, parser, or stringifier. Right now, PostCSS does nothing. Pick plugins for your case on https://www.postcss.parts/ and use them in postcss.config.js.

**Comment by kristerkari** [comment]
> maybe we should **not** silence the jest/no-standalone-expect errors in jest-setup.js?

I originally took the code from stylelint's repo and did not want to change it, so that's why it's there. I haven't checked if their test runner code has changed since then.

**Comment by kristerkari** [comment]
> Oh, and also, it would be nice if we could find a way to fix these warnings too

Those tests were missing the `parser` setting for PostCSS. Fixed now.

**Comment by XhmikosR** [comment]
I think we could cherry pick a couple of patches like the postcss warnings fix and the missing it block patches. Do you want me to do sherry pick those to master? If so, I'll rebase this branch after that.

**Comment by kristerkari** [comment]
> Do you want me to do sherry pick those to master? If so, I'll rebase this branch after that.

Please do :)

**Comment by XhmikosR** [comment]
Done and branches rebased. Just make sure you don't override them.

If you want to release a new patch from master, now it's a good time. Just make sure you do `npm i` after you switch to master, and then your usual steps.

**Comment by kristerkari** [comment]
Alright, I'll publish a patch with the changes in master.

**Comment by XhmikosR** [comment]
Added back the `node/no-unsupported-features` rules with the version set to `10.0.0`.

I think we can merge this in the v4-dev branch if you agree.

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -11,8 +11,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        # switch to `8` when https://github.com/actions/setup-node/issues/27 is fixed
-        node: [8.16.2, 10, 12, 14]
+        node: [10, 12, 14]
```
Comment (XhmikosR) [inline .github/workflows/test.yml:14]: One last thing @kristerkari: this is the latest 10.x version. The same goes for the package.json engines version. I guess it's OK since we have the ESLint rules in place, but it's your call.

Comment (kristerkari) [reply .github/workflows/test.yml:14]: That is ok. We shouldn't need to care about minor versions.
